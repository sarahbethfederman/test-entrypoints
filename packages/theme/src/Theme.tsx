import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
