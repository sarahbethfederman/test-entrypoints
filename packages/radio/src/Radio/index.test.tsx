import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import { Radio } from './index';
import { Wrapper, RadioLabel, RadioWrapper, defaultBorder } from './index.style';

let mockOnChange: jest.Mock<{}>;
let element: any;

function render(props) {
  mockOnChange = jest.fn();

  element = mount(
    <Theme>
      <Radio value="1" label="LUI single Radio Button" onChange={mockOnChange} {...props} />
    </Theme>
  );
}

describe('radioButton', () => {
  it('it should render whole component', () => {
    render({});
    expect(element.find(Radio)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(RadioLabel)).toHaveLength(1);
    expect(element.find(RadioWrapper)).toHaveLength(1);
  });

  it('should have the correct checked property based on passed props', () => {
    render({ isChecked: false });
    expect(element.find(Wrapper).props().checked).toBeFalsy();
    expect(element.find(RadioWrapper).props().checked).toBeFalsy();

    render({ isChecked: true });
    expect(element.find(Wrapper).props().checked).toBeTruthy();
    expect(element.find(RadioWrapper).props().checked).toBeTruthy();

    render({ isDisabled: true });
    expect(element.find(Wrapper).props().disabled).toBeTruthy();
    expect(element.find(RadioWrapper).props().disabled).toBeTruthy();
  });

  it('it should render correct styles when boxed', () => {
    render({ isBoxed: true });

    const testEl = element.find(Wrapper);
    expect(testEl).toHaveStyleRule(defaultBorder);
  });

  it('it should render correct styles when disabled', () => {
    render({ isDisabled: true });

    const testEl = element.find(Radio);
    expect(testEl).toMatchSnapshot();
  });
});
