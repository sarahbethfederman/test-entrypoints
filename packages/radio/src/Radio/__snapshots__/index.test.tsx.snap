// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radioButton it should render correct styles when disabled 1`] = `
.c3 {
  margin-top: 0;
  margin-bottom: 0;
  color: #c8c8c8;
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: calc(3 * var(--lendi-ui-size));
  border-radius: 6px;
}

.c0:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

.c2 {
  display: inline-block;
  margin-left: calc(0.75 * var(--lendi-ui-size));
}

.c1 {
  cursor: pointer;
  position: relative;
  display: inline-block;
  height: 32px;
  width: 32px;
  margin-left: calc(0.75 * var(--lendi-ui-size));
}

.c1:before {
  content: '';
  width: 32px;
  height: 32px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  background-color: #ffffff;
  border: 1px solid #c8c8c8;
  border-radius: 20px;
  box-sizing: border-box;
}

.c1:active::before {
  border: 1px solid #009984;
}

.c1:focus:before {
  border: 2px solid #66d9c9;
}

.c1:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

<Component
  isDisabled={true}
  label="LUI single Radio Button"
  onChange={[MockFunction]}
  value="1"
>
  <styled.label
    checked={false}
    disabled={true}
    isBoxed={false}
  >
    <label
      checked={false}
      className="c0"
      disabled={true}
    >
      <styled.input
        checked={false}
        disabled={true}
        onChange={[MockFunction]}
        type="radio"
        value="1"
      >
        <input
          checked={false}
          className="c1"
          disabled={true}
          onChange={[MockFunction]}
          type="radio"
          value="1"
        />
      </styled.input>
      <Styled(styled.p)
        color="shade.200"
        size="md"
      >
        <styled.p
          className="c2"
          color="shade.200"
          size="md"
        >
          <p
            className="c2 c3"
            color="shade.200"
            size="md"
          >
            LUI single Radio Button
          </p>
        </styled.p>
      </Styled(styled.p)>
    </label>
  </styled.label>
</Component>
`;
